examples/gateway-parallel.rs

```diff
@@ -19,37 +11,25 @@ async fn main() -> anyhow::Result<()> {
 
     let token = env::var("DISCORD_TOKEN")?;
     let client = Client::new(token.clone());
-    let config = Config::new(token.clone(), Intents::GUILDS);
-
-    let tasks = thread::available_parallelism()?.get();
+    let config = Config::new(token, Intents::GUILDS);
 
-    // Split shards into a vec of `tasks` vecs of shards.
-    let init = iter::repeat_with(Vec::new)
-        .take(tasks)
-        .collect::<Vec<Vec<_>>>();
     let shards = stream::create_recommended(&client, config, |_, builder| builder.build())
         .await?
-        .enumerate()
-        .fold(init, |mut fold, (idx, shard)| {
-            fold[idx % tasks].push(shard);
-            fold
-        });
+        .collect::<Vec<_>>();
```

Comment:
This allocation shouldn't be necessary, we can just use the iterator when spawning the tasks unless I'm missing something